internal/platform/storage/badger/badger.go

```diff
@@ -156,3 +158,66 @@ func (t Tx) GetKeysWithPrefix(prefix []byte, reverse ...bool) [][]byte {
 
 	return keys
 }
+
+func (t Tx) GetValuesWithPrefix(prefix []byte, reverse ...bool) [][]byte {
+	var values [][]byte
+
+	// Determine the reverse setting based on the optional parameter
+	isReverse := false
+	if len(reverse) > 0 {
+		isReverse = reverse[0]
+	}
+
+	opts := badger.DefaultIteratorOptions
+	opts.PrefetchValues = true
+	opts.Reverse = isReverse
+	opts.PrefetchSize = 100
+	iterator := t.tx.NewIterator(opts)
+	defer iterator.Close()
+
+	// Append 0xff to the prefix for the seek key if in reverse mode
+	var seekPrefix []byte
+	if isReverse {
```

Comment:
why do we need this reverse mode?

Reply:
it is not used right now, but we will need it eventually